Packs/SplunkPy/Integrations/SplunkPy/SplunkPy_test.py

```diff
@@ -468,3 +455,434 @@ def test_fetch_incidents(mocker):
 def test_create_mapping_dict():
     mapping_dict = splunk.create_mapping_dict(SPLUNK_RESULTS, type_field='source')
     assert mapping_dict == EXPECTED_OUTPUT
+
+
+def test_fetch_incidents(mocker):
+    mocker.patch.object(demisto, 'incidents')
+    mocker.patch.object(demisto, 'setLastRun')
+    mock_last_run = {'time': '2018-10-24T14:13:20'}
+    mock_params = {'fetchQuery': "something"}
+    mocker.patch('demistomock.getLastRun', return_value=mock_last_run)
+    mocker.patch('demistomock.params', return_value=mock_params)
+    service = mocker.patch('splunklib.client.connect', return_value=None)
+    mocker.patch('splunklib.results.ResultsReader', return_value=SAMPLE_RESPONSE)
+    splunk.fetch_incidents(service)
+    incidents = demisto.incidents.call_args[0][0]
+    assert demisto.incidents.call_count == 1
+    assert len(incidents) == 1
+    assert incidents[0]["name"] == "Endpoint - Recurring Malware Infection - Rule : Endpoint - " \
+                                   "Recurring Malware Infection - Rule"
+
+
+def test_remove_old_incident_ids():
+    """
+    Given:
+    - An array containing an ID of an incident that occurred less than an hour ago,
+    and one that occurred more than an hour ago
+
+    When:
+    - Running "remove_old_incident_ids" to remove the IDs of older incidents
+
+    Then:
+    - The ID of the incident that occurred less than an hour ago remained.
+    - The ID of the incident that occurred more than an hour ago was removed.
+    """
+    from SplunkPy import remove_old_incident_ids
+    cur_time = int(time.time())
+    incident_ids = {
+        "incident_under_one_hour_old": cur_time - 300,
+        "incident_over_one_hour_old": cur_time - 200000
+    }
+
+    assert "incident_under_one_hour_old" in incident_ids
+    assert "incident_over_one_hour_old" in incident_ids
+
+    new_incident_ids = remove_old_incident_ids(incident_ids, cur_time)
+
+    assert "incident_under_one_hour_old" in new_incident_ids
+    assert "incident_over_one_hour_old" not in new_incident_ids
+
+
+occurred_time = str(int(time.time()) - 300)
+
+first_incident = {
+    'rawJSON': '{"_raw": "first incident"}',
+    'occurred': occurred_time
+}
+
+second_incident = {
+    'rawJSON': '{"_raw": "second incident"}',
+    'occurred': occurred_time
+}
+
+
+def test_create_incident_custom_id_creates_different_ids():
+    """
+    Given:
+    - Two different incidents
+
+    When:
+    - Creating a custom ID for the incidents using "create_incident_custom_id"
+
+    Then:
+    - The IDs of the two incidents are unique.
+    """
+    from SplunkPy import create_incident_custom_id
+    first_incident_custom_id = create_incident_custom_id(first_incident)
+    second_incident_custom_id = create_incident_custom_id(second_incident)
+    assert first_incident_custom_id != second_incident_custom_id
+
+
+def test_create_incident_custom_id_length():
```

Comment:
I think it is not relevant because its an md5 so you can just test it is 32 characters, or if you take mt suggestion, 32 + 1 + len(occurred) that should also be const.